src/streaming/controllers/BufferController.js

```diff
@@ -152,8 +153,10 @@ function BufferController(config) {
             errHandler.mediaSourceError('Error creating ' + type + ' source buffer.');
         }
 
+        quality = requiredQuality === -1 ? 0 : requiredQuality;
```

Comment:
@zangue We will need to figure this out a bit more before we pull in based on the last few emails.  Maybe we can move the -1 conversation into this PR for tracking?